Fix smtchecker callback tests and add Eldarica (#565)
ethereum/solc-js: README.md

}
```

Since version 0.5.1, the ``smtSolver`` callback function is used to solve SMT queries generated by
Solidity's SMTChecker.  If you have an SMT solver installed locally, it can
be used to solve the given queries, where the callback must synchronously
return either an error or the result from the solver.  A default
``smtSolver`` callback is included in this package via the module
``smtchecker.js`` which exports the ``smtCallback`` function that takes 1) a
function that takes queries and returns the solving result, and 2) a solver

> configuration object. The module ``smtsolver.js`` has a few pre defined solver

Reviewer: ```suggestion
configuration object. The module ``smtsolver.js`` has a few predefined solver
```

configurations, and relies on Z3, Eldarica or CVC4 being installed locally.  It
exports the list of locally found solvers and a function that invokes a given
solver.

The API of the SMT callback is **experimental** and can change at any time.
The last change was in version 0.8.11.

#### Example usage with smtSolver callback

```javascript
var solc = require('solc');
const smtchecker = require('solc/smtchecker');
const smtsolver = require('solc/smtsolver');
// Note that this example only works via node and not in the browser.

var input = {
  language: 'Solidity',
  sources: {
    'test.sol': {
      content: 'contract C { function f(uint x) public { assert(x > 0); } }'
    }
  },
  settings: {
    modelChecker: {
      engine: "chc",
      solvers: [ "smtlib2" ]
    }
  }
};

var output = JSON.parse(
  solc.compile(
    JSON.stringify(input),

>     { smtSolver: smtchecker.smtCallback(smtsolver.smtSolver, smtsolver.availableSolvers[0]) }

Reviewer: Maybe we should show an example of how to pass in a custom function as a callback?

Otherwise, if user is expected to use `smtCallback()`, why don't we just expect its parameters instead of a full callback?

Author: It's just way more convenient, there's 3 layers of different stuff here

Reviewer: Then maybe we should show both. I mean, this is the only reference doc we have for the library. Having an easy to run example is good but we should also say something about what the callback looks like. Looking quickly at README, I don't think it even mentions what args the actual callback takes. Only the args of this helper.

Author: Ok, makes sense. Next PR tho? :D

Reviewer: Fine. Let's get this one merged :)

> Solidity's SMTChecker.  If you have an SMT solver installed locally, it can

Reviewer 2: Should state what solc-js version this started to work and leave a giant *experimental* disclaimer here that the API can change.

be used to solve the given queries, where the callback must synchronously
return either an error or the result from the solver.  A default
``smtSolver`` callback is distributed by ``solc-js`` via the module
``smtchecker.js`` which exports the ``smtCallback`` function that takes 1) a
function that takes queries and returns the solving result, and 2) a solver
configuration object. The module ``smtsolver.js`` has a few pre defined solver
configurations, and relies on Z3, Eldarica or CVC4 being installed locally.  It
exports the list of locally found solvers and a function that invokes a given
solver.

#### Example usage with smtSolver callback

```javascript
var solc = require('solc');

> const smtchecker = require('smtchecker.js');

Reviewer 2: ```suggestion
const smtchecker = require('solc/smtchecker');
```

> const smtsolver = require('smtsolver');

Reviewer 2: ```suggestion
const smtsolver = require('solc/smtsolver');
```
